File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int n=s.length();
        unordered_map<int,int>mp;
        char ans;

        for(auto& a:s){
            mp[a]++;
        }
        for(int i=0;i<n;++i){
            if(mp[s[i]]==1) return i;
        }
      return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int n=s.length();
        char[] arr= s.toCharArray();
        HashMap<Character,Integer>mp=new HashMap<>();
        for(int i=0;i<n;++i){
        mp.put(arr[i],mp.getOrDefault(arr[i],0)+1);
        }

        for(int i=0;i<n;++i){
            if(mp.get(arr[i])==1){
                return i;
            }
        }
        return -1;
    }
}